include/kframework/lemmas/bytes-simplification.k

```diff
@@ -68,71 +68,168 @@ module BYTES-SIMPLIFICATION [symbolic]
 
     // #range
 
-    rule #range(_, _, K) => .Bytes requires K <=Int 0 [simplification]
+    rule [range-empty]:
+      #range(_:Bytes, S:Int, W:Int) => .Bytes
+      requires S <Int 0 orBool W <=Int 0
+      [simplification(30)]
+
+    rule [range-full]:
+      #range(B:Bytes, 0, W:Int) => B
+      requires lengthBytes(B) ==Int W
+      [simplification]
+
+    rule [range-overfull]:
+      #range(B:Bytes, 0, W:Int) => B +Bytes #buf(W -Int lengthBytes(B), 0)
+      requires lengthBytes(B) <Int W
+      [simplification]
```

Comment:
I guess these two lemmas could be combined with one for `lengthBytes(B) <=Int W`. Any reason not to?

Reply:
Hmm, yes, they should be fine together, given that if we know the equality, the tail zeros will simplify out.

Reply:
The case where this would matter is if you have a variable `W` that you know `0 <=Int W` but not whether `0 ==Int W` or `0 <Int W`. In this case, having one lemma cover `0 <=Int W` will allow it to apply, but the backend will not case split on the two separate lemmas.